File: src/main/java/com/xudong/vam/core/extractor/VarExtractor.java
package com.xudong.vam.core.extractor;

import com.xudong.vam.core.model.domain.Package;

import java.io.IOException;
import java.nio.file.Path;
import java.util.List;
import java.util.concurrent.CompletableFuture;

public interface VarExtractor {
    Package extract(Path packagePath) throws IOException;

    List<CompletableFuture<Package>> extractAll(Path packagePath) throws IOException;
}

File: src/main/java/com/xudong/vam/core/extractor/VarExtractor.kt
package com.xudong.vam.core.extractor

import com.xudong.vam.core.model.domain.Package
import java.nio.file.Path
import java.util.concurrent.CompletableFuture

interface VarExtractor {
    fun extract(packagePath: Path): Package

    fun extractAll(packagePath: Path): List<CompletableFuture<Package>>
}
